services/communitytokens/estimations.go

```diff
@@ -0,0 +1,435 @@
+package communitytokens
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/status-im/status-go/contracts/community-tokens/assets"
+	"github.com/status-im/status-go/contracts/community-tokens/collectibles"
+	communitytokendeployer "github.com/status-im/status-go/contracts/community-tokens/deployer"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/services/wallet"
+	"github.com/status-im/status-go/services/wallet/bigint"
+	"github.com/status-im/status-go/transactions"
+)
+
+type CommunityTokenFees struct {
+	GasUnits      uint64                `json:"gasUnits"`
+	SuggestedFees *wallet.SuggestedFees `json:"suggestedFees"`
+}
+
+func weiToGwei(val *big.Int) *big.Float {
+	result := new(big.Float)
+	result.SetInt(val)
+
+	unit := new(big.Int)
+	unit.SetInt64(params.GWei)
+
+	return result.Quo(result, new(big.Float).SetInt(unit))
+}
+
+func gweiToWei(val *big.Float) *big.Int {
+	res, _ := new(big.Float).Mul(val, big.NewFloat(1000000000)).Int(nil)
+	return res
+}
+
+func (s *Service) deployOwnerTokenEstimate(ctx context.Context, chainID uint64, fromAddress string,
```

Comment:
The most important changes are in this file. For every operation:
- compute gas units,
- get suggested fees,
- compute l1 fee.